packages/forklift-console-plugin/src/modules/Plans/views/details/utils/hasPipelineCompleted.tsx

```diff
@@ -0,0 +1,34 @@
+import { V1beta1PlanStatusMigrationVmsPipeline } from '@kubev2v/types';
+
+/**
+ * Check if a given migration's pipeline step has completed successfully.
+ *
+ * @param {V1beta1PlanStatusMigrationVmsPipeline} pipeline - A given migration's pipeline step
+ * @returns {boolean} - True if the migration step has completed successfully, false otherwise.
+ */
+const hasPipelineCompleted = (pipeline: V1beta1PlanStatusMigrationVmsPipeline) => {
+  if (pipeline?.error) {
+    return false;
+  }
+
+  switch (pipeline?.phase) {
+    case 'Completed':
+      return true;
+    case 'Pending':
+    case 'Running':
+    default:
+      return false;
+  }
+};
+
+/**
+ *  Check if the task isn't marked as finished, but its contained pipeline step has completed successfully.
+ *
+ * @param {string } taskPhase A given task phase
+ * @param {V1beta1PlanStatusMigrationVmsPipeline} pipeline - A given migration's pipeline step which includes the task
+ * @returns {boolean} - Returns True if the task isn't marked as finished, but its contained pipeline step has completed successfully.
+ */
+export const isTaskCompletedByItsPipeline = (
+  taskPhase: string,
+  pipeline: V1beta1PlanStatusMigrationVmsPipeline,
+) => taskPhase === undefined && hasPipelineCompleted(pipeline);
```

Comment:
```suggestion
) => taskPhase === 'Completed' || (taskPhase === undefined && hasPipelineCompleted(pipeline));
```

the title of this method is `isTaskCompleted` wthout testing for `c.phase === 'Completed' ` the title is wrong.

Reply:
Your suggestion is to re-declare the scope of this utility method to indicate if task was completed (hiding  the "workaround" suggested here). And renaming as a result...

The original idea was to use this utility only for the "workaround" - for me it seems a clearer code to separate between regular valid completion to this completion by parent pipeline.

Reply:
I understand the focus on the workaround, but I’m concerned it might lead to unclear code in the long run.

To keep things clean and maintainable, it might be better to have a `isTaskCompleted` method that fully checks if a task is complete, and then separate the task, `taskPhase === 'Completed'` for handling the backend's expected behavior and `hasPipelineCompleted` for the workaround. This way, the code stays clearer and easier to update in the future.

What do you think?

Reply:
Sounds good to me